tokio/src/sync/mpsc/chan.rs

```diff
@@ -293,6 +295,85 @@ impl<T, S: Semaphore> Rx<T, S> {
         })
     }
 
+    /// Receives values into `buffer` up to its capacity
+    ///
+    /// For `limit` > 0, receives up to limit values into `buffer`.
+    /// For `limit` = 0, receives at most `super::BLOCK_CAP` values.
+    pub(crate) fn recv_many(
+        &mut self,
+        cx: &mut Context<'_>,
+        buffer: &mut Vec<T>,
+        limit: usize,
+    ) -> Poll<usize> {
+        use super::block::Read;
+
+        let mut to_go = if limit > 0 { limit } else { super::BLOCK_CAP };
```

Comment:
This `to_go` variable seems weird to me. Why not use a `maximum_length` instead?

Reply:
I have renamed `to_go` to `remaining` in the latest push -- `to_go` suggests work to be done.  The decrement loop is set up that way because I think someone will pass `usize::max_value()` to `limit` with `buffer.length() > 0` length and this makes the function's behavior simple and clear.